### PR TITLE
improve(finalizer): Optimize lookback windows

### PR DESCRIPTION
**Pull request by nicholaspai**
A lot of the finalizer slowdown currently is querying withdrawl status information about TokenBridgedEvents. Optimize the lookback settings with the following changes
- OpStack: Don't submit proofs for any transaction older than 1 day. Don't try to withdraw any withdrawal earlier than 6.5 days old
- Polygon: Don't try to withdraw anything younger than 0.75 days
- ZkSync: Don't try to withdraw anything younger than 2 days

### CODE REVIEW

**Review by james-a-morris — COMMENTED** [review]
This makes sense +1

**Review by james-a-morris — APPROVED** [review]

File: src/finalizer/utils/opStack.ts [hunk 1]
```diff
@@ -44,10 +47,20 @@ export async function opStackFinalizer(
...
   const { recentTokensBridgedEvents = [], olderTokensBridgedEvents = [] } = groupBy(
     spokePoolClient.getTokensBridged(),
-    (e) => (e.blockNumber >= earliestBlockToProve ? "recentTokensBridgedEvents" : "olderTokensBridgedEvents")
+    (e) => {
+      if (e.blockNumber >= earliestBlockToProve) {
+        return "recentTokensBridgedEvents";
+      } else if (e.blockNumber < latestBlockToFinalize) {
+        return "olderTokensBridgedEvents";
+      }
+    }
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply src/finalizer/utils/opStack.ts:63]: No the second filter was added to ignore submitting potential proofs for withdrawals sent more than one day ago!

[the remark this answers is not in the document]
Comment (james-a-morris) [reply src/finalizer/utils/opStack.ts:63]: Right but we could've embedded it in the ternary.

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**